test-suite/js/src/test/scala/org/scalajs/testsuite/compiler/RuntimeTypesTest.scala

```diff
@@ -64,13 +64,12 @@ class RuntimeTypesTest {
     assumeTrue("Assumed compliant asInstanceOf", hasCompliantAsInstanceOfs)
     def test(x: Any): Unit = {
       try {
-        classOf[Nothing].cast(x)
+        val result = classOf[Nothing].cast(x)
         fail("casting " + x + " to Nothing did not fail")
+        identity(result) // discard `result` without warning
       } catch {
-        case th: Throwable =>
-          assertTrue(th.isInstanceOf[ClassCastException])
-          assertEquals(x + " is not an instance of scala.runtime.Nothing$",
-              th.getMessage)
+        case th: ClassCastException =>
+          // ok
```

Comment:
If this is only about the type, should we use `ExpectedException` on the test?

Reply:
If you mean `assertThrows`, that doesn't quite work in this case. Because we need to test that the exception is thrown at the very point of the assignment to `result`, and not any later. If we do a simple `assertThrows`, we cannot check that. In particular we cannot check that the error is thrown before the statement with `identity`.

Reply:
No, I meant JUnit's `ExpectedException`. But indeed, that's a fair point. I didn't get the subtlety about the assignment being necessary.